### ISSUE TITLE
multi-gpus training（without ignite）converges slower

### ISSUE THREAD

**Issue by Zhao-zw** [michuanhaohao/reid-strong-baseline#140]
At first，thank you.
I use multi-gpus training（without ignite） code. I  use one gpu training it , that results obtained with DDP + SyncBN are definitely worse than the original implementation. It  converges slower is the main problems.
I checked the model,dataloader,optimizer ,I'm sure it has the same training config with original one.
Do you know what caused it？
Could you give me some suggestions?Thank you very much!

**Comment by L1aoXingyu** [comment]
If you train with DDP, you need to write a custom distributed data sampler to make each process load a subset of the original dataset that is exclusive to it.
Besides, if you training with triplet loss, you can only mine pairs in each process.

**Comment by Zhao-zw** [comment]
> 如果使用DDP进行训练，则需要编写一个自定义的分布式数据采样器，以使每个进程都加载原始数据集的专有子集。
> 此外，如果您进行三重态损失训练，则每个过程只能挖掘一对。

I‘m sorry，I checked the code. When i use one gpu training it，the training didn't use the DDP+ SyncBN.
the code as following:
        else:
            # Single GPU model
            self.model.cuda()
            self.optim = make_optimizer(cfg, self.model, num_gpus)
            self.scheduler = WarmupMultiStepLR(self.optim, cfg.SOLVER.STEPS, cfg.SOLVER.GAMMA,
                                          cfg.SOLVER.WARMUP_FACTOR,
                                          cfg.SOLVER.WARMUP_ITERS, cfg.SOLVER.WARMUP_METHOD)
            self.scheduler.step()
            self.mix_precision = False
            if cfg.SOLVER.FP16:
                # Single model using FP16
                self.model, self.optim = amp.initialize(self.model, self.optim,
                                                        opt_level='O1')
                self.mix_precision = True
                self.logger.info('Using fp16 training')
            self.logger.info('Trainer Built')
            return

**Comment by L1aoXingyu** [comment]
I don't know your problem. you train with one GPU but cannot get the same result as the repo, right?

**Comment by Zhao-zw** [comment]
> I don't know your problem. you train with one GPU but cannot get the same result as the repo, right?

yes,and it converges slower than repo.
